lib/spack/spack/s3_handler.py

```diff
@@ -44,7 +44,7 @@ def __getattr__(self, key):
 
 def _s3_open(url):
     parsed = url_util.parse(url)
-    s3 = s3_util.create_s3_session(parsed, connection=s3_util.get_mirror_connection(parsed))
```

Comment:
Was it a bug that this didn't pass `url_type="fetch"` previously?

Reply:
I think so yes